pkg/gpu/nvidia/manager.go

```diff
@@ -339,5 +448,18 @@ func (ngm *nvidiaGPUManager) Stop() error {
 	}
 	ngm.stop <- true
 	<-ngm.stop
+	if ngm.healthCheckRunning {
+		ngm.healthCheckStop <- true
+		<-ngm.healthCheckStop
+	}
```

Comment:
Should we close ngm.health channel as well?

Reply:
Yeah good catch, even though the behavior will not change (i.e. `ListAndWatch()` will be blocked and never update devices with kubelet), i think it will be better to close it here. 

And even if we close the channel here, I think it is fine to have `ListAndWatch()` call `sendDevice()` when the channel is closed.